viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    new_movie = {}
+    
+    if not title or not genre or not rating:
+        return None
+    
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"] += [movie]
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"] += [movie]
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    watchlist = user_data["watchlist"]
+
+    for i in range(len(watchlist)):
+        if title == watchlist[i]["title"]:
+            user_data["watched"] += [watchlist[i]]
+            watchlist.pop(i)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    watched = user_data["watched"]
+
+    if not watched:
+        return 0
+    
+    sum = 0
+
+    for i in range(len(watched)):
+        sum += watched[i]["rating"]
+
+    # average = sum(values) / len(user_data)
+    avg_rating = sum / len(watched)
+
+    return avg_rating
+
+def get_most_watched_genre(user_data):
+    pass
+    # What is deemed most popular? By rating? By count?
+    watched = user_data["watched"]
+
+    genre_dict = {}
+
+    if len(watched) == 0:
+        return None
+    
+    # watched[0]["genre"] == "Fantasy"
+    for movie in watched:
+        genre_dict[movie["genre"]] = 1
+        
+    for movie in watched:
+        if genre_dict[movie["genre"]] >= 1:
+            genre_dict[movie["genre"]] += 1
+
+
+    # max(d, key = d.get)
+    most_watched = max(genre_dict, key = genre_dict.get)
+    
+    return most_watched
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+# Consider the movies that the user has watched, 
+# and consider the movies that their friends have watched. 
+# Determine which movies the user has watched, 
+# but none of their friends have watched.
+# Return a list of dictionaries, that represents a list of movies
+
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+
+    user_unique_movies = []
+
+    friends_movie_watched = []
+
+    # Big O notation of O(n^2), try to refactor
+    for friend in friends:
+        for i in range(len(friend["watched"])):
+            friends_movie_watched.append(friend["watched"][i]) 
+
+    for movie in user_watched:
+        if movie not in friends_movie_watched:
+            if movie not in user_unique_movies:

```

Comment:
So this would give you a time complexity of `O(n^3)` since you have `not in` which has a time complexity of `O(n)` itself. Is there a way you could implement a helper function in order to decrease the time complexity of your wave 3 functions?